training/src/main/java/wolox/training/models/Book.java

```diff
@@ -47,15 +49,13 @@ public long getId() {
         return id;
     }
 
-    public void setId(long id) {
-        this.id = id;
-    }
-
     public String getTitle() {
         return title;
     }
 
     public void setTitle(String title) {
+        Preconditions.checkNotNull(title);
```

Comment:
Merge both like this, everywhere:
Preconditions.checkArgument(title != null and !title.isEmpty());